serve.go

```diff
@@ -767,24 +771,27 @@ func (c *config) depClient() (dep.Client, error) {
 	return client, nil
 }
 
-func (c *config) setupDEPSync() {
+func (c *config) setupDEPSync(logger log.Logger) depsync.Syncer {
 	if c.err != nil {
-		return
+		return nil
 	}
 
 	client, err := c.depClient()
 	if err != nil {
 		c.err = err
-		return
+		return nil
 	}
 	var opts []depsync.Option
+	logger = log.With(logger, "component", "depsync")
 	if client != nil {
-		opts = append(opts, depsync.WithClient(client))
+		opts = append(opts, depsync.WithClient(client), depsync.WithLogger(logger))
 	}
-	_, c.err = depsync.New(c.pubclient, c.db, opts...)
+	var syncer depsync.Syncer
+	syncer, c.err = depsync.New(c.pubclient, c.db, logger, opts...)
 	if err != nil {
-		return
+		return nil
```

Comment:
log this err before returning nil.

Reply:
It is logged, actually. The `c.err` is assigned to the config object and handled elsewhere. Most of the `setup*()` are odd like this. That _said_ there is a bug where I should have been checking `c.err` and not `err`!